bootstrap/api/v1alpha1/rke2config_types.go

```diff
@@ -134,6 +134,11 @@ type RKE2AgentConfig struct {
 	// AirGapped is a boolean value to define if the bootstrapping should be air-gapped,
 	// basically supposing that online container registries and RKE2 install scripts are not reachable.
 	AirGapped bool `json:"airGapped,omitempty"`
+
+	// AdditionalCloudInit is a field that allows users to specify additional cloud-init configuration to be included in the
+	// generated cloud-init script.
+	//+optional
+	AdditionalCloudInit string `json:"additionalCloudInit,omitempty"`
```

Comment:
I think we should probably choose a different name for this to make it a bit more generic. The reason being is that when we add support for Ignition we may want to allow a similar functionality to provide additional configuration. Perhaps something like **AdditionalBootstrapData** or something like that.

Reply:
@alexander-demicev the thing is : that `AdditionalConfig` field in Kubeadm is already inside a `IgnitionSpec` struct, so it is not generic, it is ignition specific. I thought I would be more specific to avoid misunderstandings.

Reply:
@richardcase usually, this would be referred to as `userData` maybe I should put `AdditionalUserData`

Reply:
in that case, it makes sense to name it `AdditionalCloudInit` so people don't confuse it with IgnitionSpec. Alternatively, we can add a new structure  `AdditionalUserData` that contains 2 fields inside: `IgnitionConfig`, `CloudInitConfig`. Why is API design so hard? :) I would prefer to avoid having to specify additional cloud-init config in one place and ignition in another nested object.

Reply:
I suggest I change it to `AdditionalUserData` for now, and we can think about the details of ignition later. What do you think ?

Reply:
If I understand it right, we might ignore the way Kubeadm does it, and do it this way:
- Do not use a nested config for Ignition
- Use `AdditionalUserData` as a free text field which might contain Ignition or Cloud-Init configuration
- Figure out a way to make the user understand what he is doing (an enum field for choosing between cloud-init or ignition?)
- We might want to validate the `AdditionalUserData` field by parsing it for signs of ignition or cloud-init.

Reply:
I like this idea, that way we don't need 2 different fieldsfor the same thing